### DISCUSSION TITLE
Check if default wallet source is expired before building payment

### DISCUSSION THREAD

**Discussion by mdamian322** [solidusio/solidus#3980] [New Features or Ideas]
We were having an issue of payments getting created with a default card that was expired. We ended up overriding the `build` method in the [DefaultPaymentBuilder Model](https://github.com/solidusio/solidus/blob/v2.6/core/app/models/spree/wallet/default_payment_builder.rb) and just added `expired?` to the check on [line 17](https://github.com/solidusio/solidus/blob/v2.6/core/app/models/spree/wallet/default_payment_builder.rb#L17).  So the method looks like:
```
def build
 default = order.user&.wallet&.default_wallet_payment_source
  if default && !default.payment_source.expired? && order.payments.where(source_type: default.payment_source_type).none?
      Spree::Payment.new(
        payment_method: default.payment_source.payment_method,
        source: default.payment_source
      )
  end
 end
  ```
Any thoughts?

**Comment by aldesantis** [comment]
@mdamian322 what is the payment source class you're using? I don't see an `expired?` method in the abstract `Spree::PaymentSource` class or in `Spree::CreditCard`, so I must assume this is a custom implementation?
